usr/src/uts/common/fs/zfs/sys/dnode.h

```diff
@@ -136,7 +136,13 @@ typedef struct dnode_phys {
 	uint8_t dn_flags;		/* DNODE_FLAG_* */
 	uint16_t dn_datablkszsec;	/* data block size in 512b sectors */
 	uint16_t dn_bonuslen;		/* length of dn_bonus */
-	uint8_t dn_pad2[4];
+	/*
+	 * crypto is ready to encrypt dnodes, but until large-dnode
+	 * commit is merged, dn_extra_slots is left here as a
+	 * place-holder, but assumed to be 0 at all times.
+	 */
```

Comment:
As you say, to keep it simpler for large-dnodes to merge in future. I will update the comment with the next round of patches.